### DISCUSSION TITLE
Filament filename generation

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#9887] [Help]
Hello all,

Filament generates file name with this format "01HG62QP7GRG0X1Y43ZJ7NFSK7.jpg"

I'm using Spatie Media Library and I'm importing media from url. I would like to generate the same "file name format" than Filament but I can't find how Filament handle the name generation in the source code.

`$model->addMediaFromUrl($image->url)->toMediaCollection('images', 'private')->withVisibility('private');`

Any idea?

**Comment by leandrocfe** [comment]
Did you try [getUploadedFileNameForStorageUsing](https://filamentphp.com/docs/3.x/forms/fields/file-upload#generating-custom-file-names) ?
